ocs_ci/utility/utils.py

```diff
@@ -2062,23 +2062,47 @@ def convert_device_size(unformatted_size, units_to_covert_to):
         if units == 'Ti':
             return absolute_size
         elif units == 'Gi':
-            return absolute_size * 1024
+            return absolute_size / 1024
         elif units == 'Mi':
-            return absolute_size * 1024 * 1024
+            return absolute_size / 1024 / 1024
+        elif units == 'Ki':
+            return absolute_size / 1024 / 1024 / 1024
     elif units_to_covert_to == 'GB':
         if units == 'Ti':
-            return absolute_size / 1024
+            return absolute_size * 1024
         elif units == 'Gi':
             return absolute_size
         elif units == 'Mi':
-            return absolute_size * 1024
+            return absolute_size / 1024
+        elif units == 'Ki':
+            return absolute_size / 1024 / 1024
     elif units_to_covert_to == 'MB':
         if units == 'Ti':
-            return absolute_size / 1024 / 1024
+            return absolute_size * 1024 * 1024
         elif units == 'Gi':
+            return absolute_size * 1024
+        elif units == 'Mi':
+            return absolute_size
+        elif units == 'Ki':
             return absolute_size / 1024
+    elif units_to_covert_to == 'KB':
+        if units == 'Ti':
+            return absolute_size * 1024 * 1024 * 1024
+        elif units == 'Gi':
+            return absolute_size * 1024 * 1024
         elif units == 'Mi':
+            return absolute_size * 1024
+        elif units == 'Ki':
             return absolute_size
+    elif units_to_covert_to == 'B':
+        if units == 'Ti':
+            return absolute_size * 1024 * 1024 * 1024 * 1024
+        elif units == 'Gi':
+            return absolute_size * 1024 * 1024 * 1024
+        elif units == 'Mi':
+            return absolute_size * 1024 * 1024
+        elif units == 'Ki':
+            return absolute_size * 1024
```

Comment:
Can use dictionary?
Dict={(units_to_covert_to,units):2^20}
return absolute_size* Dict[(units_to_covert_to,units)]

Reply:
good point. I will change to a dict